### DISCUSSION TITLE
Extract Redux DevTools support to decouple from UI?

### DISCUSSION THREAD

**Discussion by quark-zju** [jotaijs/jotai-devtools#121] [General]
Hi, I found the `useAtomsDevtools`  pretty powerful and `jotai-devtools` seems to have some dependencies on the UI part. I wonder if it is a good idea to extract the Redux integration to a separate package that have less UI dependencies?

From https://github.com/jotaijs/jotai-devtools/discussions/68 it seems `jotai-devtools/utils` already exist and maybe that can be published as a separate npm package?

**Comment by arjunvegda** [comment]
Hey @quark-zju 

Thanks for reaching out! @dai-shi and I touched upon this idea of mono repos in the early days of `jotai-devtools`, keeping it as part of a single package with different entry points was a deliberate decision. 

At some point (if time permits 😓), I want to address the concerns of "extra" dependencies. I aim to move away from `@mantine/*` and `@emotion/react` to something like [shadcn ui](https://github.com/shadcn-ui/ui). This should alleviate the pain for our react-native/expo users. Would you like to help us on this?

**Reply by dai-shi** [reply]
At that point, I didn't expect there would be so many dependencies.
Making jotai-devtools lightweight would be good anyways, but if you want, having a separate repo `jotai-redux-devtools` (or something) is okay for me. (and we can re-report it as `jotai-devtools/utils` for backward compatibility.)

**Comment by quark-zju** [comment]
Thanks for the context! I was thinking about:
- `jotai-devtools` remains unchanged. i.e. it does not depend on `jotai-redux-devtools`.
- `jotai-redux-devtools` is a new simplified package that provides the Redux DevTools integration.
- Their source code is in a same repo.

It seems the `jotai-redux-devtools` would need 0 dependencies. There is an import on `@redux-devtools/extension` but it seems empty and can be removed: https://github.com/jotaijs/jotai-devtools/blob/7c38133a360c0c97db6406becdc1bf939e1001e7/src/utils/types.ts#L1

So even if the UI part is moved to other library the new library and `react-*` dependencies are still extra dependencies to projects that do not use them, comparing with 0-dependency redux devtools.

**Reply by dai-shi** [reply]
We would like to avoid monorepo. Other than that, it sounds good. Would you like to lead the `jotai-redux-devtools` project?

**Reply by quark-zju** [reply]
If there will be multiple repos or packages, I have 2 questions:

1. Naming. Looking at the features, maybe `jotai-debug-hooks` or `jotai-devtools-hooks` is a better name? It's not just for redux.
2. Repo. I noticed the Redux part was moved out from `jotai/` initially. Given that it's 400-line 0-dependency it might make sense to move back (maybe as `debug-hooks/` so the name does not conflict with `devtools`)? I notice that there are commits like [this](https://github.com/jotaijs/jotai-devtools/commit/c925680c00dfba205e9e1456476d02e2ada39c67) that changed both the hooks and UI so it seems having them in a same repo would be convenient (otherwise moving back to jotai or a separate repo seems equally inconvenient for cross hook/UI changes, which I'm not sure how often it happens).

**Reply by dai-shi** [reply]
I'm fine with any names. Moving out from `jotai` means we want a new maintainer. That's how we make the core small and organize jotaijs/*. Sounds like you are more comfortable with `jotai-devtools/utils` with more lightweight dependencies.

**Reply by quark-zju** [reply]
I see. Yes, `jotai-devtools/utils` with no dependencies would work in my case. I can have a look updating this repo to provide a way to provide the `jotai-devtools/utils` package, if this is the direction to go.
